File: mobile/lib/core/models/doctor.dart
/// Doctor model
class Doctor {
  final String id;
  final String name;
  final String? specialization;
  final String? qualification;
  final String? registrationNumber;
  final int? experienceYears;
  final String? phone;
  final String? email;
  final String clinicId;
  final double consultationFee;
  final double? followupFee;
  final int slotDuration;
  final Map<String, dynamic>? workingHours;
  final String? photoUrl;
  final List<String>? languages;
  final bool isActive;
  final bool acceptingNewPatients;

  Doctor({
    required this.id,
    required this.name,
    this.specialization,
    this.qualification,
    this.registrationNumber,
    this.experienceYears,
    this.phone,
    this.email,
    required this.clinicId,
    required this.consultationFee,
    this.followupFee,
    required this.slotDuration,
    this.workingHours,
    this.photoUrl,
    this.languages,
    required this.isActive,
    required this.acceptingNewPatients,
  });

  factory Doctor.fromJson(Map<String, dynamic> json) {
    return Doctor(
      id: json['id'],
      name: json['name'],
      specialization: json['specialization'],
      qualification: json['qualification'],
      registrationNumber: json['registration_number'],
      experienceYears: json['experience_years'],
      phone: json['phone'],
      email: json['email'],
      clinicId: json['clinic_id'],
      consultationFee: (json['consultation_fee'] ?? 500).toDouble(),
      followupFee: json['followup_fee']?.toDouble(),
      slotDuration: json['slot_duration'] ?? 15,
      workingHours: json['working_hours'],
      photoUrl: json['photo_url'],
      languages: json['languages'] != null
          ? List<String>.from(json['languages'])
          : null,
      isActive: json['is_active'] ?? true,
      acceptingNewPatients: json['accepting_new_patients'] ?? true,
    );
  }

  String get feeDisplay => '₹${consultationFee.toStringAsFixed(0)}';

  String get experienceDisplay {
    if (experienceYears == null) return '';
    return '$experienceYears years';
  }
}

File: backend/app/models/doctor.py
"""
Doctor model for medical professionals.
"""

import uuid
from decimal import Decimal
from typing import TYPE_CHECKING, Any

from sqlalchemy import Boolean, ForeignKey, Integer, Numeric, String, Text
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.models.base import BaseModel
from app.models.types import JSONB

if TYPE_CHECKING:
    from app.models.appointment import Appointment
    from app.models.clinic import Clinic
    from app.models.lab_result import LabOrder


class Doctor(BaseModel):
    """
    Doctor model for medical professionals.

    Attributes:
        name: Full name with title (Dr. Priya Patel)
        specialization: Medical specialization
        qualification: Degrees (MBBS, MD, etc.)
        registration_number: Medical council registration
        clinic_id: Associated clinic
        consultation_fee: Default consultation fee
        slot_duration: Default appointment duration in minutes
        working_hours: JSON defining weekly schedule
        is_active: Whether doctor is accepting appointments
    """

    __tablename__ = "doctors"

    # Professional Info
    name: Mapped[str] = mapped_column(String(200), nullable=False)
    specialization: Mapped[str | None] = mapped_column(String(100), nullable=True)
    qualification: Mapped[str | None] = mapped_column(String(500), nullable=True)
    registration_number: Mapped[str | None] = mapped_column(String(50), nullable=True)
    experience_years: Mapped[int | None] = mapped_column(Integer, nullable=True)

    # Contact
    phone: Mapped[str | None] = mapped_column(String(15), nullable=True)
    email: Mapped[str | None] = mapped_column(String(255), nullable=True)

    # Clinic Association
    clinic_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("clinics.id"),
        nullable=False,
        index=True,
    )

    # Practice Settings
    consultation_fee: Mapped[Decimal] = mapped_column(
        Numeric(10, 2),
        default=Decimal("500.00"),
    )
    followup_fee: Mapped[Decimal | None] = mapped_column(Numeric(10, 2), nullable=True)
    slot_duration: Mapped[int] = mapped_column(Integer, default=15)  # minutes

    # Working Hours - JSON format:
    # {
    #   "monday": [{"start": "09:00", "end": "13:00"}, {"start": "17:00", "end": "20:00"}],
    #   "tuesday": [...],
    #   "sunday": null  // Off day
    # }
    working_hours: Mapped[dict[str, Any] | None] = mapped_column(JSONB, nullable=True)

    # Break slots - for lunch, etc.
    break_slots: Mapped[list[dict[str, str]] | None] = mapped_column(JSONB, nullable=True)

    # Profile
    bio: Mapped[str | None] = mapped_column(Text, nullable=True)
    photo_url: Mapped[str | None] = mapped_column(String(500), nullable=True)
    languages: Mapped[list[str] | None] = mapped_column(JSONB, nullable=True)

    # Status
    is_active: Mapped[bool] = mapped_column(Boolean, default=True)
    accepting_new_patients: Mapped[bool] = mapped_column(Boolean, default=True)

    # Relationships
    clinic: Mapped["Clinic"] = relationship("Clinic", back_populates="doctors")
    appointments: Mapped[list["Appointment"]] = relationship(
        "Appointment",
        back_populates="doctor",
    )
    lab_orders: Mapped[list["LabOrder"]] = relationship(
        "LabOrder",
        back_populates="doctor",
    )

    def __repr__(self) -> str:
        return f"<Doctor {self.name} ({self.specialization})>"
